File: code/community/1298788959849025538/docker-websocket-connection-deepgram.cs
using System;
using System.Net.WebSockets;
using System.Threading;
using System.Threading.Tasks;
using System.Text;

class Program
{
    static async Task Main()
    {
        // Get the Deepgram API Key from environment variables
        string apiKey = Environment.GetEnvironmentVariable("DEEPGRAM_API_KEY");
        string url = "wss://api.deepgram.com/v1/listen";
        
        using (ClientWebSocket client = new ClientWebSocket())
        {
            client.Options.SetRequestHeader("Authorization", $"Token {apiKey}");

            // Connect to the Deepgram WebSocket API
            await client.ConnectAsync(new Uri(url), CancellationToken.None);
            Console.WriteLine("Connected to Deepgram API");

            // Sending example data/message through WebSocket
            byte[] buffer = Encoding.UTF8.GetBytes("Your data or message here");
            await client.SendAsync(new ArraySegment<byte>(buffer), WebSocketMessageType.Text, true, CancellationToken.None);
            
            // Logic to handle receiving messages can be added here
        }
    }
}

// Note: Ensure DEEPGRAM_API_KEY is set in the environment variables beforehand.

File: code/community/1298788959849025538/docker-websocket-connection-deepgram.go
package main

import (
	"os"
	"log"
	"github.com/gorilla/websocket"
)

func main() {
	// Retrieve the API key from the environment variable
	apiKey := os.Getenv("DEEPGRAM_API_KEY")
	
	url := "wss://api.deepgram.com/v1/listen"

	// Prepare headers for the WebSocket connection
	headers := map[string][]string{
		"Authorization": {"Token " + apiKey},
	}

	// Connect to the WebSocket
	c, _, err := websocket.DefaultDialer.Dial(url, headers)
	if err != nil {
		log.Fatal("Failed to connect to Deepgram API:", err)
	}
	defer c.Close()

	log.Println("Connected to Deepgram API")

	// Example: Send data/message
	err = c.WriteMessage(websocket.TextMessage, []byte("Your data or message here"))
	if err != nil {
		log.Fatal("Failed to send message:", err)
	}

	// Additional logic for handling WebSocket communication (receiving messages) can be added here
}

// Note: Make sure to set the environment variable DEEPGRAM_API_KEY correctly.